src/components/RecipeView.vue

```diff
@@ -223,6 +239,20 @@ export default {
         }
     },
     methods: {
+        escapeHtml: function(unsafeString) {
+            return unsafeString
+                .replace(/&/g, "&amp;")
+                .replace(/\~/g, "&#732;")
+                .replace(/</g, "&lt;")
+                .replace(/>/g, "&gt;")
+                .replace(/"/g, "&quot;")
+                .replace(/'/g, "&#039;");
+        },
+        convertRecipeReferences: function(text) {
+            let re = /(^|\s)#r\/(\d+)(\s|$)/g
```

Comment:
This will break when two links are just adjacent (aka `#r/123 #r/124`). Here a [look-ahead](https://www.stefanjudis.com/today-i-learned/the-complicated-syntax-of-lookaheads-in-javascript-regular-expressions/) might be needed.

Reply:
I’m really struggling with this 😓 

This won’t work
`(?<=^|\s)(^|\s)#r\/(\d+)(?=$|\s)(\s|$)`

This detects the occurrences but doesn’t capture the first space (if present) and breaks newlines
`(?<=^|\s)#r\/(\d+)(?=$|\s)(\s|$)`

any ideas @christianlupus?

Reply:
Actually, the second one works. I forgot that Markdown needs an empty line before it renders a line break 😄